manage_containers.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/python3
+
+"""
+This tool manages containers for building the Linux kernel with various compilers
+"""
+
+import os
+import subprocess
+import sys
+import argparse
+
+supported_compilers = ['gcc-4.9', 'gcc-5', 'gcc-6', 'gcc-7', 'gcc-8', 'gcc-9',
+                       'gcc-10', 'gcc-11', 'gcc-12', 'gcc-13', 'gcc-14',
+                       'clang-5', 'clang-6', 'clang-7', 'clang-8',
+                       'clang-9', 'clang-10', 'clang-11', 'clang-12',
+                       'clang-13', 'clang-14', 'clang-15', 'clang-16', 'clang-17',
+                       'all']
+
+class Container:
+    """
+    Represents a Docker container configured for building the Linux kernel with a specified compiler
+
+
+    Class Attributes:
+        sudo_wrapper (str): wrap commands with sudo (if needed)
+        quiet (bool): enable quiet option in Docker build
+
+    Instance Attributes:
+        gcc (str): Version of GCC compiler
+        clang (str): Version of Clang compiler
+        ubuntu (str): Version of Ubuntu
+        id (str): Docker image ID
+    """
+
+    sudo_wrapper = ''
+    quiet = False
+
+    def __init__(self, gcc_version, clang_version, ubuntu_version):
+        self.gcc = gcc_version
+        self.clang = clang_version
+        self.ubuntu = ubuntu_version
+        self.id = self.check()
+
+    def add(self):
+        """Build the Docker container with the specified compilers"""
+        build_args = ['--build-arg', f'GCC_VERSION={self.gcc}',
+                      '--build-arg', f'CLANG_VERSION={self.clang}',
+                      '--build-arg', f'UBUNTU_VERSION={self.ubuntu}',
+                      '--build-arg', f'UNAME={os.getlogin()}',
+                      '--build-arg', f'UID={os.getuid()}',
+                      '--build-arg', f'GID={os.getgid()}',
+                      '-t', f'kernel-build-container:gcc-{self.gcc}',
+                      '-t', f'kernel-build-container:clang-{self.clang}'
+        ]
+
+        if self.quiet:
+            build_args += ['-q']
+        subprocess.run([self.sudo_wrapper, 'docker', 'build', *build_args, '.'],
+                        text=True, check=True)
+        self.check()
+
+    def rm(self):
+        """Remove the Docker container if it exists and not running"""
+        running = subprocess.run(f'{Container.sudo_wrapper} docker ps | '
+                                 f'grep -E "kernel-build-container:'
+                                 f'(gcc-{self.gcc}|clang-{self.clang})" || true',
```

Comment:
It's better to use a list instead of a large string here.

Reply:
Agree.
Maybe it's better to parse the results of `docker ps` here in python script to avoid this long command.

Reply:
better now?

https://github.com/a13xp0p0v/kernel-build-containers/pull/22/commits/76a512ca88e5b286ac4240aa94c6a0b26cae04aa